src/pretalx/submission/models/submission.py

```diff
@@ -361,31 +405,16 @@ def cancel(self, person=None, force=False, orga=True):
         self._set_state(SubmissionStates.CANCELED, force, person=person)
         self.log_action('pretalx.submission.cancel', person=person, orga=True)
 
-        from pretalx.schedule.models import TalkSlot
-
-        TalkSlot.objects.filter(
-            submission=self, schedule=self.event.wip_schedule
-        ).delete()
-
     def withdraw(self, person=None, force=False, orga=False):
         self._set_state(SubmissionStates.WITHDRAWN, force, person=person)
-        from pretalx.schedule.models import TalkSlot
-
-        TalkSlot.objects.filter(
-            submission=self, schedule=self.event.wip_schedule
-        ).delete()
         self.log_action('pretalx.submission.withdraw', person=person, orga=orga)
 
     def remove(self, person=None, force=False, orga=True):
         self._set_state(SubmissionStates.DELETED, force, person=person)
+        self.log_action('pretalx.submission.deleted', person=person, orga=True)
```

Comment:
The ordering here was intentional, in case this action fails and is run without an active transaction.